commands.js

```diff
@@ -5,8 +5,201 @@
+ *
+ * this.canBroadcast()
+ *   Signifies that a message can be broadcast, as long as the user
+ *   has permission to. This will check to see if the user used
+ *   "!command" instead of "/command". If so, it will check to see
+ *   if the user has permission to broadcast (by default, voice+ can),
+ *   and return false if not. Otherwise, it will add the message to
+ *   the room, and turn on the flag this.broadcasting, so that
+ *   this.sendReply and this.sendReplyBox will broadcast to the room
+ *   instead of just the user that used the command.
+ *
+ *   Should usually be near the top of the command, like:
+ *     if (!this.canBroadcast()) return false;
+ *
+ * this.canBroadcast(suppressMessage)
+ *   Functionally the same as this.canBroadcast(). However, it
+ *   will look as if the user had written the text suppressMessage.
+ *
+ * this.canTalk()
+ *   Checks to see if the user can speak in the room. Returns false
+ *   if the user can't speak (is muted, the room has modchat on, etc),
+ *   or true otherwise.
+ *
+ *   Should usually be near the top of the command, like:
+ *     if (!this.canTalk()) return false;
+ *
+ * this.canTalk(message, room)
+ *   Checks to see if the user can say the message in the room.
+ *   If a room is not specified, it will default to the current one.
+ *   If it has a falsy value, the check won't be attached to any room.
+ *   In addition to running the checks from this.canTalk(), it also
+ *   checks to see if the message has any banned words, is too long,
+ *   or was just sent by the user. Returns the filtered message, or a
+ *   falsy value if the user can't speak.
+ *
+ *   Should usually be near the top of the command, like:
+ *     target = this.canTalk(target);
+ *     if (!target) return false;
+ *
+ * this.parse(message, inNamespace)
+ *   Runs the message as if the user had typed it in.
+ *
+ *   Mostly useful for giving help messages, like for commands that
+ *   require a target:
+ *     if (!target) return this.parse('/help msg');
+ *
+ *   If `inNamespace` is true, then the message is parsed in that
+ *   corresponding namespace:
+ *     // command msg is in namespace test. (ie. /test msg)
+ *     this.parse('/help', true); // is parsed as if the user said
+ *                                // '/test help'
+ *
+ *   After 10 levels of recursion (calling this.parse from a command
+ *   called by this.parse from a command called by this.parse etc)
+ *   we will assume it's a bug in your command and error out.
+ *
+ * this.targetUserOrSelf(target, exactName)
+ *   If target is blank, returns the user that sent the message.
+ *   Otherwise, returns the user with the username in target, or
+ *   a falsy value if no user with that username exists.
+ *   By default, this will track users across name changes. However,
+ *   if exactName is true, it will enforce exact matches.
+ *
+ * this.getLastIdOf(user)
+ *   Returns the last userid of an specified user.
```

Comment:
This is a pretty bad explanation, compared to all the other explanations which are good because I wrote them myself. :P

A good explanation would be something like:

> Returns a user's most recently used userid. If a user isn't a guest, this works the same way as user.userid, but if the user is a guest (i.e. has logged out), this returns the user's most recent non-guest userid.